### PR TITLE
Force the proper Qute language for files under resources/templates

### LINKED ISSUE #450: Automatically force qute language of files under src/main/resources/templates of Quarkus projects
Nobody actually uses the qute.[html|txt|json|yaml] suffixes, especially in all the quarkus/qute examples. 
We should try to make our users' life easier by automatically switching to the proper language id, files under src/main/resources/templates of Quarkus projects

Similar to https://github.com/redhat-developer/vscode-quarkus/blob/master/src/extension.ts#L96

### PR DESCRIPTION
**Pull request by fbricon**
Fixes #450

Signed-off-by: Fred Bricon <fbricon@gmail.com>

### CODE REVIEW

**Review by AlexXuChen — COMMENTED** [review]
Looks like it's working as intended, given the following project directory:
```
src
├── main
    ├── other
        ├──a.html
    ├── resources
        ├── b.html
        ├── templates
            ├── nested_resource
                ├── c.html
            ├── d.html
```
Only `c.html` and `d.html` are recognized as `Qute HTML` files, while `a.html` and `b.html` are considered `HTML` files. This works `txt`, `yaml` and `json`, and doesn't apply to any unsupported file extension, such as `py`, `java`, etc..

File: src/extension.ts [hunk 1]
```diff
@@ -112,30 +113,45 @@ function tryToForceLanguageId(document: TextDocument, fileName: string, document
...
+              + `to be '${languageId}' in order to provide language support. `
+              + `This behavior can be configured in settings.`, DISABLE_IN_SETTINGS, CONFIGURE_IN_SETTINGS);
+            response.then(result => {
+              if (result === CONFIGURE_IN_SETTINGS) {
+                commands.executeCommand('workbench.action.openSettings', QuarkusConfig.PROPERTIES_LANGUAGE_MISMATCH);
+              } else if (result === DISABLE_IN_SETTINGS) {
+                QuarkusConfig.setPropertiesLanguageMismatch(PropertiesLanguageMismatch.ignore).then(() => {
+                  languages.setTextDocumentLanguage(document, oldLanguageId);
+                });
+                context.globalState.update(QuarkusConfig.QUARKUS_SET_LANGUAGE_ID, 'true');
```
Comment (rgrunber) [inline src/extension.ts]: So if the user chooses `CONFIGURE_IN_SETTINGS`, the popup still gets shown every time ? Based on the name I figured it should stop showing in both cases.

File: src/extension.ts [hunk 2]
```diff
@@ -112,30 +113,45 @@ function tryToForceLanguageId(document: TextDocument, fileName: string, document
         const oldLanguageId: string = document.languageId;
         languages.setTextDocumentLanguage(document, languageId);
         if (notifyUser && !documentCache.includes(document.fileName)) {
```
Comment (rgrunber) [inline src/extension.ts:115]: I'm confused about this line. I know it wasn't touched, but with this change `updateLanguageId(..)` always sets its last parameter as `false`. This means that `tryToForceLanguageId` will have the `notifyUser` parameter always `false`. So the question is, when does this code below run, if `notifyUser` is always `false`.

Am I missing some case ?

**Update** : I guess now that I think of it, I'm fine with this staying as long as we confirm we want to keep the functionality, even though it may not be used.

File: src/extension.ts (hunk 1 above)
Comment (AlexXuChen) [reply src/extension.ts]: Should we make it more apparent with a "Don't Show Again" option?

Comment (rgrunber) [reply src/extension.ts]: No, I think it should be automatic. Whichever option you click, the popup shouldn't be shown again.

**Comment by AlexXuChen** [comment]
The pop-up:
Screenshot from 2022-01-27 16-40-38

**Comment by rgrunber** [comment]
Ok, just move `context.globalState.update(QuarkusConfig.QUARKUS_SET_LANGUAGE_ID, 'true');` [back to where it was](https://github.com/redhat-developer/vscode-quarkus/compare/f170558a19a1d615da3b20703425f5285a8366c4..93f09c457fc904b7a2e202f093e2e6b06173b08c). I realized that if we don't, then the `documentCache` logic becomes useless.

After that, I think this is ready to squash+merge.

File: src/extension.ts [hunk 3]
```diff
@@ -162,5 +178,13 @@ async function updateLanguageId(document: TextDocument, documentCache: string[],
       return;
     }
     tryToForceLanguageId(document, fileName, documentCache, propertiesLanguageMismatch, 'yaml', onExtensionLoad);
+  } else if (document.fileName.includes(`resources${path.sep}templates${path.sep}`)) {
```
Comment (AlexXuChen) [inline src/extension.ts:181]: After testing, it appears this is platform independent as required

File: package.json [hunk 4]
```diff
@@ -75,6 +75,7 @@
     "languages": [
       {
         "id": "quarkus-properties",
+
```
Comment (fbricon) [inline package.json]: unnecessary change

File: src/QuarkusConfig.ts [hunk 5]
```diff
@@ -31,6 +31,7 @@ export namespace QuarkusConfig {
   export const ALWAYS_SHOW_WELCOME_PAGE = 'quarkus.tools.alwaysShowWelcomePage';
   export const DEBUG_TERMINATE_ON_EXIT = 'quarkus.tools.debug.terminateProcessOnExit';
   export const PROPERTIES_LANGUAGE_MISMATCH = "quarkus.tools.propertiesLanguageMismatch";
+  export const QUARKUS_SET_LANGUAGE_ID = 'quarkus.tools.set.language.id';
```
Comment (fbricon) [inline src/QuarkusConfig.ts]: QUARKUS_OVERRIDE_LANGUAGE_ID = 'quarkus.tools.override.languageId';
